### PR TITLE
test: Improves unit tests for ColumnType + QA fixes

### PR DESCRIPTION
**Pull request by ttannis**
### Summary of Changes
This PR completes our code complete criteria for the targeted feature branch.
1. Updates `parser.ts` logic to cover an edge case in Presto types. (Commented as `Case 3.1` in code, case  is included in unit tests)
2. Updates styles based on design QA
3. Support for logging click interaction on the column type button.

### Tests

1.  Adds improved test coverage for `parser.ts`, using similarly complicated structures found in production.
2. Adds improves test coverage for `ColummnType`

### Documentation

N/A for this feature.

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes, including screenshots of any UI changes.
- [x] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/index.tsx [hunk 1]
```diff
@@ -127,8 +131,8 @@ export class ColumnType extends React.Component<
         >
           <Modal.Header closeButton>
             <Modal.Title>
-              {MODAL_TITLE}
-              <div className="column-name">{columnName}</div>
+              <div className="main-title">{MODAL_TITLE}</div>
```
Comment (Golodhros) [inline amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/index.tsx:134]: Not sure of how this displays, but we might try to use a lower hierarchy h here?

File: amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/parser.ts [hunk 2]
```diff
@@ -116,6 +133,11 @@ export function parseNestedType(
   columnType: string,
   databaseId: string
 ): NestedType | null {
+  // Presto includes un-needed "" characters
+  if (databaseId === 'presto') {
```
Comment (Golodhros) [inline amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/parser.ts:137]: Nit: maybe a constant for 'presto' ?

File: amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/styles.scss [hunk 3]
```diff
@@ -56,13 +54,12 @@ $modal-dialog-width: 418px;
     /* Override react-bootstrap styles to match design */
     padding: $spacer-3 !important;
 
-    /* A design audit of modals is needed to reconcile differences accumulated over time */
-    .modal-title {
-      font-family: $font-family-body !important;
+    .main-title {
+      @extend %text-title-w1;
```
Comment (Golodhros) [inline amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/styles.scss:58]: Yay!

File: amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/styles.scss [hunk 4]
```diff
@@ -56,13 +54,12 @@ $modal-dialog-width: 418px;
...
     }
-
-    .column-name {
-      font-size: $w2-font-size;
+    .sub-title {
+      @extend %text-subtitle-w3;
+      color: $text-secondary;
```
Comment (Golodhros) [inline amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/styles.scss:62]: Nit: Not super sure, but usually linters warn if we don't let a space after extends or mixins.

File: amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/index.tsx (hunk 1 above)
Comment (ttannis) [reply amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/index.tsx:134]: Let me take a look to see if one matches.

File: amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/styles.scss (hunk 4 above)
Comment (ttannis) [reply amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/styles.scss:62]: Since we still have so many I haven't been noticing if my code has increased those warnings, let me check.

File: amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/index.tsx (hunk 1 above)
Comment (ttannis) [reply amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/index.tsx:134]: The `<Modal.Title>` from `react-bootstrap` is already an `h4`, I will use `h5` and it doesn't affect the display since we have no applicable styles on that.

File: amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/styles.scss (hunk 4 above)
Comment (ttannis) [reply amundsen_application/static/js/components/TableDetail/ColumnListItem/ColumnType/styles.scss:62]: I didn't see a warning on this, but fixed some in these files while I was looking.
